File: fp/fp.dart
main(List<String> args) {
  var numbers = [1, 2, 3];

  print("filter method");
  print(numbers.where((e) => e > 1));

  print("map method");
  print(numbers.map((e) => e.toString()));

  print("map method - hash type list");
  var h = [
    {"name": 'A'},
    {"name": 'B'},
    {"name": 'C'}
  ];
  print(h.map((e) => e["name"]));

  print("reduce method");
  print(numbers.reduce((a, b) => a + b));
}

File: fp/fp.js
log = console.log;

let numbers = [1, 2, 3];

log('filter method');
log(numbers.filter((e) => e > 1));

log('map method');
log(numbers.map((e) => e.toString()));

log('map method - hash type list');
d = [{ name: 'A' }, { name: 'B' }, { name: 'C' }];
log(d.map((e) => e.name));

log('reduce method');
log(numbers.reduce((a, b) => a + b));
